src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -972,10 +974,31 @@ private synchronized void reset(boolean wipeData) throws IOException {
 
     @Override
     public synchronized void afterTest() throws IOException {
+        assertShardIndexCounter();
         wipeDataDirectories();
         randomlyResetClients(); /* reset all clients - each test gets its own client based on the Random instance created above. */
     }
 
+    /**
+     * Wipes any data that a test can leave behind: indices, templates and repositories
+     */
+    public void wipe() {
+        assertShardIndexCounter();
+        super.wipe();
+    }
+
+    private void assertShardIndexCounter() {
```

Comment:
assertAfterTest() is called after wipe() (https://github.com/elastic/elasticsearch/blob/master/src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java#L657 ) so there would be indices left to check if I moved it there